Patron Deity Rework

james9017 opened the pull request:
- Reworked "select_patron_deity_germanic_decision" for each of the faiths in germanic_religion.
- Decision and Event (religious_decision.0102 from Vanilla) now uses dynamic localisation for each deity positon
- Unique modifiers have had to be made for each of deities of each Germanic faith (as modifiers don't work with dynamic loc), though this opens up the option of changing the descriptions/effects of the modifiers if needed
- Removed Ullr and Freyr options, with the intention to add more fitting options at a later
- Added Seaxneat option exclusively for Saxon cultured characters of any Germanic faith. This replaces Tyr as their War God.
- Added new "set_deity_effect", as replacing the vanilla counterpart was causing my game to crash
- Modified faith_conversion_events to account for the changes made, but this does not seem to have worked. When characters convert, they retain the modifier, even after unpausing for a while.
- Misc. changes to "Patron Deity" localisation locations.

james9017 committed:
Patron Deity Rework - Draft 1

- Reworked "select_patron_deity_germanic_decision" for each of the faiths in germanic_religion.
- Decision and Event (religious_decision.0102 from Vanilla) now uses dynamic localisation for each deity positon
- Unique modifiers have had to be made for each of deities of each Germanic faith (as modifiers don't work with dynamic loc), though this opens up the option of changing the descriptions/effects of the modifiers if needed
- Removed Ullr and Freyr options, with the intention to add more fitting options at a later
- Added Seaxneat option exclusively for Saxon cultured characters of any Germanic faith. This replaces Tyr as their War God.
- Added new "set_deity_effect", as replacing the vanilla counterpart was causing my game to crash
- Modified faith_conversion_events to account for the changes made, but this does not seem to have worked. When characters convert, they retain the modifier, even after unpausing for a while.
- Misc. changes to "Patron Deity" localisation locations.

DC123456789 commented on Britannia/common/decisions/BTWK_religious_decisions.txt:
```diff
@@ -0,0 +1,241 @@
+			has_character_modifier = BTWK_gothic_deity_odin_modifier
+			has_character_modifier = BTWK_gothic_deity_tyr_modifier
+			has_character_modifier = BTWK_gothic_deity_thor_modifier
+
```
Spacing?

james9017 replied:
Whitespace at the bottom was just a mistake. Put spacing between the difference faiths' modifiers just to make it easier to read.

DC123456789 commented on Britannia/events/religious_decision_events.txt:
```diff
@@ -0,0 +1,2878 @@
+	# 	set_bhakti_effect = { BHAKTI = bhakti_germanic_generic_freyr }
+	# }
+
+	# NORSE
```
May be easier to combine the different faith variants of the same god into a single option and just use if/else blocks... up to you though

james9017 replied:
I could do that, though I was intending to keep all the faiths together rather than the same god options together. I'm also not too sure how I'd do this for the random_list for the decision tooltip. Use triggered_descs?

james9017 committed:
Patron Deity Rework - Draft 2

- Organised options etc. by God, rather than by faith, to make it easier to add more gods in the future.

james9017 committed:
Patron Deity Rework - Draft 3 (Need Help)

- Grouped new modifiers by deity in faith_conversion.0002.
- Intention was to remove the modifiers if the character changes religion, else interchange deity "x" from faith one's name to faith two's name.
- This was working before I regrouped the if-statements by deity (originally they were grouped by faith & worked fine). I'd like it to work this way instead to make it easier to add new deities to the list as discussed.
- I've checked for my brackets to see if I've indented wrong, but there's nothing obvious.
- I've reviewed the on_action which triggers this, and root is definitely the character, if that helps debugging.
-Nothing regarding this file in the error log either
- I'm 99% sure that the issue is not that I've used "religion = ..." rather than "has_religion" in the limit, as the latter breaks the religious_decision_events file.

james9017 committed:
Patron Deity Rework - Ready for Review

- Moved overwritten vanilla files into correct directory. Everything now seems to work as expected. Ready to look at merging into master if all is okay.
- Can look at adding a few more gods before merge. One for Health/Fertility definitely, one for stewardship/construction, maybe one for an 'evil god' if they're a witch or something?
